### PR TITLE
Feature/erc20 mintable local faucet

### PR DESCRIPTION
**Pull request by tabaktoni**
## Usage related changes

- Resolve https://github.com/Shard-Labs/starknet-devnet/issues/122

## Development related changes

- Introduced mint_lite for qucker mint without transaction and mint with transaction
## Checklist:

- [x] No linter errors
- [ ] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

**Review by FabijanC — CHANGES_REQUESTED** [review]
1. Let's keep the liteness only configurable through the HTTP request (so as it is, no CLI args).
2. Can you try minifying ERC20_Mintable.json? Its size is 4.7 MB, perhaps we could reduce it by:
    - removing whitespaces
    - removing `debug_info` from the json:
        - this would reduce the size as well as remove the paths originating from the computer where it was generated

File: test/test_fee_token.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
 """Fee token related tests."""
-
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply test/test_fee_token.py:2]: I added imports and removed spaces between imports

File: test/test_fee_token.py [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+import requests
 from starkware.starknet.core.os.class_hash import compute_class_hash
 from starkware.starknet.core.os.contract_address.contract_address import calculate_contract_address_from_hash
 from starknet_devnet.fee_token import FeeToken
-from .util import assert_equal
+from .util import assert_equal, devnet_in_background
+
+ACCOUNTS_SEED_DEVNET_ARGS = [
+    "--accounts",
+    "3",
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply test/test_fee_token.py:12]: How would you test this without seed? Account addresses will be diff every time.
Why is it important not to have devnet arguments?

**Comment by yoga-braavos** [comment]
@FabijanC @tabaktoni thanks for this.
Can you please give a name, symbol and decimals to the FeeToken (maybe same as TST in alpha-goerli) so connecting Wallets will have something reasonable to show?

File: test/test_fee_token.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FabijanC) [reply test/test_fee_token.py:12]: It's not that it's important not to have devnet arguments, it's just that they serve no purpose (and may lead to confusion). Basically the address used as the target of minting doesn't need to belong to an actually deployed account.

File: starknet_devnet/blueprints/base.py [hunk 3]
```diff
@@ -94,3 +94,20 @@ def get_predeployed_accounts():
...
+    amount = request_json["amount"]
+    is_lite = request_json.get('lite', False)
+
+    if is_lite:
+        await FeeToken.mint_lite(address, amount)
+    else:
+        await FeeToken.mint(address, amount, state.starknet_wrapper)
+
+    new_balance = await FeeToken.get_balance(int(address, 16))
+    return jsonify({"new_balance": new_balance, "unit": "wei"})
```
[the remark this answers is not in the document]
Comment (FabijanC) [reply starknet_devnet/blueprints/base.py:113]: get_balance doesn't create a transaction so it shouldn't return any tx hashes. But the minting endpoint could.

**Review by FabijanC — COMMENTED** [review]
Suggestion:
- extra logging will be done in a separate PR
- log token address on startup (this was requested by users)

File: test/test_fee_token.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tabaktoni) [reply test/test_fee_token.py]: https://github.com/Shard-Labs/starknet-devnet/pull/143/commits/44e2627bd86867fdbd40170bf6e5f20261eb540b

File: test/test_fee_token.py [hunk 4]
```diff
@@ -23,3 +32,28 @@ def test_precomputed_address():
...
+
+@pytest.mark.fee_token
+@devnet_in_background(*ACCOUNTS_SEED_DEVNET_ARGS)
+def test_mint():
+    """Asert that mint will increase account balance"""
+    json_load = {
+        "address": "0x6e3205f9b7c4328f00f718fdecf56ab31acfb3cd6ffeb999dcbac41236ea502",
+        "amount": 50_000
+    }
+    response = requests.post(f"{GATEWAY_URL}/mint", json=json_load)
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply test/test_fee_token.py:37]: https://github.com/Shard-Labs/starknet-devnet/pull/143/commits/44e2627bd86867fdbd40170bf6e5f20261eb540b

File: test/test_fee_token.py [hunk 5]
```diff
@@ -23,3 +32,28 @@ def test_precomputed_address():
...
+    assert response.status_code == 200
+    assert response.json().get('new_balance') == 60000
+
+@pytest.mark.fee_token
+@devnet_in_background(*ACCOUNTS_SEED_DEVNET_ARGS)
+def test_mint_lite():
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply test/test_fee_token.py:48]: https://github.com/Shard-Labs/starknet-devnet/pull/143/commits/44e2627bd86867fdbd40170bf6e5f20261eb540b

File: starknet_devnet/fee_token.py [hunk 6]
```diff
@@ -61,23 +68,56 @@ async def deploy(cls, starknet: Starknet):
...
-            deploy_execution_info=None
+            deploy_execution_info=None,
         )
 
     @classmethod
     async def get_balance(cls, address: int) -> int:
         """Return the balance of the contract under `address`."""
         assert cls.contract
         response = await cls.contract.balanceOf(address).call()
+        print("original balance value:", response.result.balance)
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply starknet_devnet/fee_token.py]: Will be solved in separate issue

File: starknet_devnet/fee_token.py [hunk 7]
```diff
@@ -61,23 +68,56 @@ async def deploy(cls, starknet: Starknet):
...
+                str(amount_uint256.high),
+            ],
+            "signature": [],
+            "contract_address": hex(cls.ADDRESS),
+            "type": "INVOKE_FUNCTION",
+        }
+        transaction = validate_transaction(json.dumps(transaction_data).encode("utf-8"))
+        c_address, tx_hash, result = await starknet_wrapper.invoke(transaction)
+
+        print({'c_address':c_address, 'tx_hash':tx_hash, 'result':result})
```
[the remark this answers is not in the document]
Comment (tabaktoni) [reply starknet_devnet/fee_token.py]: will be resolved in separate issue for logging

File: starknet_devnet/blueprints/base.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tabaktoni) [reply starknet_devnet/blueprints/base.py]: https://github.com/Shard-Labs/starknet-devnet/pull/143/commits/2e27e7fc928bf97884e4c1fb150088aa19b84b99

File: README.md [hunk 8]
```diff
@@ -411,6 +411,30 @@ Response:
 }
 ```
 
+## Mintable
+
+Used ERC20 Mintable fee Token
+
+### Mint fee token
```
Comment (FabijanC) [inline README.md:418]: We should also have the lite option documented.

**Pull request merged**